src/pycram/robot_description.py

```diff
@@ -1,512 +1,495 @@
-        :type configuration: str
-        :param static_joint_states: List of floats/joint configurations
-        :type static_joint_states: list[float]
+        :param description: RobotDescription to register.
         """
-        if all(map(lambda n: isinstance(n, Number), static_joint_states)):
-            if len(static_joint_states) == len(self.joints):
-                configurations = list(self.static_joint_states.keys())
-                if configuration not in configurations:
-                    self.static_joint_states = self.static_joint_states
-                    self.static_joint_states[configuration] = static_joint_states
-                    return True
-                else:
-                    rospy.logwarn("(robot_description) The chain %s already has static joint values "
-                            "for the config %s.", self.name, configuration)
-            else:
-                rospy.logerr("(robot_description) The number of the joint values does not equal the amount of the joints"
-                       "for the chain %s.", self.name)
+        if description.name in self.descriptions.keys():
+            raise ValueError(f"Description {description.name} already exists")
+        self.descriptions[description.name] = description
+
 
-    def get_static_joint_chain(self, configuration: str) -> Union[None, Dict[str, float]]:
+class RobotDescription:
+    """
+    Base class of a robot description. Contains all necessary information about a robot, like the URDF, the base link,
+    the torso link and joint, the kinematic chains and cameras.
+    """
+    current_robot_description: RobotDescription = None
+
+    def __init__(self, name: str, base_link: str, torso_link: str, torso_joint: str, urdf_path: str):
         """
-        Gets a static joint chain as dictionary of its joint names and values.
+        Initialize the RobotDescription. The URDF is loaded from the given path and used as basis for the kinematic
+        chains.
 
-        :param configuration: Name of the configuration
-        :type configuration: str
-        :return: dict[str: float]
+        :param name: Name of the robot
+        :param base_link: Base link of the robot, meaning the first link in the URDF
+        :param torso_link: Torso link of the robot
+        :param torso_joint: Torso joint of the robot, this is the joint that moves the torso upwards if there is one
+        :param urdf_path: Path to the URDF file of the robot
         """
-        try:
-            joint_values = self.static_joint_states[configuration]
-        except KeyError:
-            rospy.logerr("(robot_description) Configuration %s is unknown for the chain %s.", configuration, self.name)
-            return None
-        return dict(zip(self.joints, joint_values))
+        self.name = name
+        self.base_link = base_link
+        self.torso_link = torso_link
+        self.torso_joint = torso_joint
```

Comment:
Shouldn't the class member be define outside of the init and the doc and types also there?

Reply:
for all classes in this document